scene/gui/text_edit.cpp

```diff
@@ -6056,6 +6066,24 @@ int TextEdit::get_selection_origin_column(int p_caret) const {
 	return carets[p_caret].selection.origin_column;
 }
 
+int TextEdit::get_next_composite_character_column(int p_caret) const {
+	ERR_FAIL_INDEX_V(p_caret, carets.size(), -1);
+	if (get_caret_column(p_caret) == text[get_caret_line(p_caret)].length()) {
+		return 0;
+	} else {
+		return TS->shaped_text_next_character_pos(text.get_line_data(get_caret_line(p_caret))->get_rid(), (get_caret_column(p_caret)));
+	}
+}
+
+int TextEdit::get_previous_composite_character_column(int p_caret) const {
+	ERR_FAIL_INDEX_V(p_caret, carets.size(), -1);
+	if (get_caret_column(p_caret) == 0) {
+		return text[get_caret_line(p_caret) - 1].length();
```

Comment:
This looks like it could crash if used on the first line.
If it is on the first line at column 0, it should return 0.